worlds/hk/__init__.py

```diff
@@ -441,12 +443,36 @@ def set_rules(self):
             world.completion_condition[player] = lambda state: state.count("Defeated_Pantheon_5", player)
         elif goal == Goal.option_godhome_flower:
             world.completion_condition[player] = lambda state: state.count("Godhome_Flower_Quest", player)
+        elif goal == Goal.option_grub_hunt:
+            if self.options.GrubHuntGoal == "all":
```

Comment:
```suggestion
            if self.options.GrubHuntGoal.value == GrubHuntGoal.special_range_names["all"]:
```

This is needed to function if the option is provided numerically